src/date-navigator/date-navigator.tsx

```diff
@@ -0,0 +1,115 @@
+import dayjs from "dayjs";
+import {
+    ArrowLeft,
+    ArrowRight,
+} from "../shared/internal-calendar/calendar-manager.style";
+import { CalendarHelper, DateHelper } from "../util";
+import {
+    Container,
+    HeaderArrowButton,
+    StyledDateText,
+    StyledDayText,
+    Wrapper,
+} from "./date-navigator.style";
+import { DateNavigatorProps } from "./types";
+
+export const DateNavigator = ({
+    selectedDate,
+    ...optionalProps
+}: DateNavigatorProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+    const date = DateHelper.toDayjs(selectedDate);
+    const dateText = DateHelper.toDayjs(selectedDate)
+        .format("D MMMM YYYY")
+        .toString();
+    const isToday = DateHelper.isSame(selectedDate, dayjs());
+    const dayText = isToday
+        ? "Today"
+        : DateHelper.toDayjs(selectedDate).format("dddd");
+
+    // =============================================================================
+    // EVENT HANDLERS
+    // =============================================================================
+    const isLeftArrowDisabled = () => {
+        if (!optionalProps.minDate) {
+            return false;
+        }
+        return (
+            CalendarHelper.isDisabledDay(
+                date,
+                undefined,
+                optionalProps.minDate
+            ) || DateHelper.isSame(date, optionalProps.minDate)
+        );
+    };
+
+    const isRightArrowDisabled = () => {
+        if (!optionalProps.maxDate) {
+            return false;
+        }
+        return (
+            CalendarHelper.isDisabledDay(
+                date,
+                undefined,
+                undefined,
+                optionalProps.maxDate
+            ) || DateHelper.isSame(date, optionalProps.maxDate)
+        );
+    };
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    return (
+        <Container
+            id="date-navigator-container-id"
+            className={optionalProps.className}
+            data-testid={optionalProps["data-testid"]}
+        >
+            {optionalProps.onLeftArrowClick && (
+                <HeaderArrowButton
+                    id="date-navigator-left-arrow-btn-id"
+                    data-testid="date-navigator-left-arrow-btn"
+                    disabled={optionalProps.isLoading || isLeftArrowDisabled()}
+                    focusHighlight={false}
+                    tabIndex={-1}
```

Comment:
no need to set tabIndex, it's a good practice to keep it focusable for accessibility

do include a `aria-label` prop to describe the button (e.g. "Previous day")